src/game_enemy.h

```diff
@@ -362,6 +365,10 @@ inline bool Game_Enemy::IsFlying() const {
 	return enemy->levitate;
 }
 
+inline int Game_Enemy::GetUnarmedBattleAnimationId() const {
+        return enemy->maniac_unarmed_animation;
```

Comment:
Please guard this with a Player::IsPatchManiac and return 1 when the patch is unavailable.

Reply:
Fixed.

Reply:
This should also be an easyrpg extension outside of just maniac patch, but we can worry about that later. We don't have any framework for this yet.

Reply:
Also @rueter37 Please put the conditional in `Game_Enemy` not at the call site.

Reply:
I'm probably lacking intelligence but I don't understand this one.

Reply:
He meant that you should put the Maniac-Check and the return 1 here where I marked it with my comment and in game_battlealgorithm just call enemy->GetUnarmedBattleAnimationId().
This has the advantage that any caller of this function will get the expected value.

Reply:
Thanks for the explanation! I hope I got this right now.

Reply:
yeah this way